common/quotas/global/collection/collection.go

```diff
@@ -137,14 +267,54 @@ func (c *Collection) OnStop(ctx context.Context) error {
 
 	select {
 	case <-ctx.Done():
-		return fmt.Errorf("ollection failed to stop, context canceled: %w", ctx.Err())
+		return fmt.Errorf("collection failed to stop, context canceled: %w", ctx.Err())
 	case <-c.stopped:
 		return nil
 	}
 }
 
-func (c *Collection) For(key string) Limiter {
-	return c.limits.Load(key)
+func (c *Collection) keyMode(key string) keyMode {
+	rawMode := keyMode(c.keyModes(key))
+	switch rawMode {
+	case modeLocal, modeGlobal, modeLocalShadowGlobal, modeGlobalShadowLocal, modeDisabled:
+		c.logger.Debug("ratelimiter key mode", tag.Dynamic("key", key), tag.Dynamic("mode", rawMode))
+		return rawMode
+	default:
+		c.logger.Error("ratelimiter key forcefully disabled, bad mode", tag.Dynamic("key", key), tag.Dynamic("mode", modeDisabled))
```

Comment:
nit: `key` + `mode` will be probably breaking some columnar log data somewhere

Reply:
dropped debug, it's in a fairly hot path.  error now has full tags tho